fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java

```diff
@@ -199,147 +170,112 @@ public void launchLao() {
     wallet.importSeed(seed);
     setIsWalletSetUp(true);
-    openWallet();
   }
 
   public void newSeed() {
     wallet.newSeed();
-    mOpenSeedEvent.postValue(new SingleEvent<>(true));
   }
 
   /** Getters for MutableLiveData instances declared above */
   public LiveData<List<Lao>> getLAOs() {
     return mLAOs;
   }
 
-  public LiveData<SingleEvent<String>> getOpenLaoEvent() {
-    return mOpenLaoEvent;
-  }
-
-  public LiveData<SingleEvent<Boolean>> getOpenHomeEvent() {
-    return mOpenHomeEvent;
-  }
-
-  public LiveData<SingleEvent<String>> getOpenConnectingEvent() {
-    return mOpenConnectingEvent;
-  }
-
-  public LiveData<SingleEvent<HomeViewAction>> getOpenConnectEvent() {
-    return mOpenConnectEvent;
-  }
-
-  public LiveData<SingleEvent<Boolean>> getOpenLaunchEvent() {
-    return mOpenLaunchEvent;
-  }
-
-  public LiveData<SingleEvent<Boolean>> getLaunchNewLaoEvent() {
-    return mLaunchNewLaoEvent;
-  }
-
-  public LiveData<SingleEvent<Boolean>> getCancelNewLaoEvent() {
-    return mCancelNewLaoEvent;
-  }
-
-  public Boolean isWalletSetUp() {
-    return mIsWalletSetUp.getValue();
+  public boolean isWalletSetUp() {
+    Boolean setup = mIsWalletSetUp.getValue();
+    if (setup == null) return false;
+    else return setup;
   }
 
   public LiveData<Boolean> getIsWalletSetUpEvent() {
     return mIsWalletSetUp;
   }
 
-  public LiveData<SingleEvent<Boolean>> getOpenWalletEvent() {
-    return mOpenWalletEvent;
-  }
-
-  public LiveData<SingleEvent<Boolean>> getOpenSeedEvent() {
-    return mOpenSeedEvent;
+  public void openHome(FragmentManager manager) {
+    setCurrentFragment(manager, R.id.fragment_home, HomeFragment::newInstance);
   }
 
-  public LiveData<SingleEvent<String>> getOpenLaoWalletEvent() {
-    return mOpenLaoWalletEvent;
+  public void openCameraPermission(FragmentManager manager, ActivityResultRegistry registry) {
+    setCurrentFragment(
+        manager, R.id.fragment_camera_perm, () -> CameraPermissionFragment.newInstance(registry));
   }
 
-  public LiveData<SingleEvent<Boolean>> getOpenSettingsEvent() {
-    return mOpenSettingsEvent;
-  }
-
-  public LiveData<SingleEvent<Boolean>> getOpenSocialMediaEvent() {
-    return mOpenSocialMediaEvent;
-  }
-
-  /*
-   * Methods that modify the state or post an Event to update the UI.
-   */
-
-  public void openLAO(String laoId) {
-    mOpenLaoEvent.postValue(new SingleEvent<>(laoId));
-  }
-
-  public void openHome() {
-    mOpenHomeEvent.postValue(new SingleEvent<>(true));
+  public void openLaunch(FragmentManager manager) {
+    setCurrentFragment(manager, R.id.fragment_launch, LaunchFragment::newInstance);
   }
 
   public void openConnecting(String laoId) {
-    mOpenConnectingEvent.postValue(new SingleEvent<>(laoId));
+    Intent intent = new Intent(getApplication().getApplicationContext(), ConnectingActivity.class);
+    intent.putExtra(Constants.LAO_ID_EXTRA, laoId);
+    getApplication().startActivity(intent);
```

Comment:
Did you forget to pass the activity in the arguments like the other activity start ?

Reply:
Yes 😅